snap/sysparams/sysparams.go

```diff
@@ -0,0 +1,100 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package sysparams
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+// For mocking in tests
+var (
+	osutilEnsureFileState = osutil.EnsureFileState
+)
+
+// SnapdSystemParams manages certain system configuration parameters
+// aspects like the homedirs configuration that must be available
+// for other binaries, as snap-confine.
+type SnapdSystemParams struct {
+	path     string
+	Homedirs string
+}
+
+func parseSystemParams(contents string) (*SnapdSystemParams, error) {
+	params := &SnapdSystemParams{}
+	scanner := bufio.NewScanner(strings.NewReader(contents))
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// ignore empty lines and comments
+		if len(line) == 0 || strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		if strings.HasPrefix(line, "homedirs=") {
+			params.Homedirs = strings.TrimPrefix(line, "homedirs=")
+		} else {
+			return params, fmt.Errorf("cannot parse invalid line: %s", line)
+		}
+	}
+	return params, nil
+}
+
+// Open either opens the existing file at the given path, and parses
+// the file, or in case the file does not exist, it will initialize
+// and return a new SnapdSystemParams structure.
+func Open(path string) (*SnapdSystemParams, error) {
+	if !osutil.FileExists(path) {
+		return &SnapdSystemParams{path: path}, nil
+	}
+
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	params, err := parseSystemParams(string(data))
+	if err != nil {
+		return params, err
+	}
+	params.path = path
+	return params, nil
+}
+
+// Update flushes the changes (if any) to path supplied to Open.
+func (ssp *SnapdSystemParams) Update() error {
+	contents := fmt.Sprintf("homedirs=%s\n", ssp.Homedirs)
+	desiredState := &osutil.MemoryFileState{
+		Content: []byte(contents),
+		Mode:    0644,
+	}
+	if err := osutilEnsureFileState(ssp.path, desiredState); errors.Is(err, osutil.ErrSameState) {
```

Comment:
given that this is tiny and we don't return anything about whether is was updated or not I wonder if it shouldn't simply use  osutil.AtomicWriteFile?